### PR TITLE
Add New England synthetic residential AMI dataset pipeline (15-min Faraday, six states)

### PR DESCRIPTION
**Pull request by doctorbrownphd**
## What this adds

A complete regional dataset pipeline that trains Faraday on NREL End-Use Load Profiles for the six New England states and generates a 1,000-home synthetic residential AMI dataset at 15-minute resolution, in the net-load convention (PV homes export; negative readings are real). No real household appears anywhere in the pipeline — training data is NREL ResStock simulated building stock.

The dataset card (`src/opensynth/datasets/new_england/DATASET_CARD.md`) documents provenance, method, evaluation and limitations, using the fidelity / utility / privacy vocabulary of Chai et al. 2024 (arXiv:2407.11785). The dataset itself (35M rows) is not in the repo; a Zenodo DOI will be added to the card on release.

## Pipeline

```
get-ne-data --timeseries          # EULP metadata + RECS 2020 + GHCN + 1,800 stratified buildings (manifest-pinned)
scripts/fetch_pv_shapes.py        # PVWatts-model PV shapes from Open-Meteo ERA5 2018, PVGIS-NSRDB cross-check
preprocess-ne-data                # melt -> label -> DER augmentation -> pack (defaults to packaged PV shapes)
scripts/train_ne_faraday.py       # VAE 150 epochs + GMM sweep k in {200,100,400}
generate-ne-dataset               # RECS-drawn homes, real 2018 temp trajectories, magnitude calibration
```

Four executed notebooks under `notebooks/new_england/` walk through data preparation, training, evaluation and generation.

## Key design points

- **Conditional sampling**: `NewEnglandFaradayModel.sample_gmm_conditional` conditions every GMM component on a chosen label vector in closed form (per-component Gaussian conditioning + likelihood reweighting), so generation can target (state, archetype, heating fuel, DER flags, month, day-of-week, daily temp bin). Pinned to analytic values in tests.
- **Real weather trajectories**: generated homes walk their state's actual 2018 daily temperature-bin sequence (GHCN), so cold snaps land on the right dates coherently across homes.
- **Magnitude calibration**: day-independent sampling collapses per-home annual variance; each home draws one persistent scale factor from the real within-segment relative-annual distribution (documented in the card, recorded per home in the metadata file).
- **15-min support**: `preprocess_lcl.parse_settlement_period` gained a `periods_per_hour` parameter (default preserves half-hourly LCL behaviour).

## Evaluation (full numbers in the card and executed notebook)

| Check | Result |
|---|---|
| Winter aggregate peak timing vs ISO-NE | 0 min |
| Summer aggregate peak timing vs ISO-NE | 60 min |
| Seasonal mean profile shape vs EULP holdout | r = 0.95–0.99 |
| Annual kWh vs RECS 2020 weighted mean (released dataset) | +4.1 %, KS 0.076 |
| EV evening signature | 0.466 vs 0.354 energy share |
| PV midday net-load depression | −0.30 kWh/15 min |
| TSTR season classification (framework utility) | accuracy delta 0.002 |

Distributional weaknesses are reported honestly (ACF and peak-count KS reflect the known day-independence of Faraday sampling) and the framework sections of `ne_evaluation.ipynb` reuse the repo's existing `seasonal_statistics`/`autocorrelation` tooling.

## Changes touching existing code (all backward compatible, called out for review)

- `datasets_utils.download_data`: wget dependency replaced with stdlib urllib (browser UA for hosts that 403), now atomic via `.part` + rename.
- `LCLData`/`LCLDataModule`: optional `feature_cols` parameter and a `dataset_cls` hook; defaults preserve existing behaviour byte-for-byte, and the NE data module is a thin subclass instead of a copy.
- `split_household_ids`: optional `seed` parameter (default keeps the historical module-seeded behaviour).
- `GaussianMixtureModel.update_params`: normalises the `nll` buffer to its registered `[1]` shape so saved state_dicts load into a fresh module (previously saved 0-dim and failed `load_state_dict`).
- `preprocess_lcl.parse_settlement_period(periods_per_hour)` as above.
- CLI: `get-ne-data`, `preprocess-ne-data`, `generate-ne-dataset` commands; small fix to the LCL download function name.

## Testing

`pytest`: 519 passed, 3 xfailed. New tests cover the sampling math (analytic pins), timestamp conventions (EULP period-ending EST), download atomicity/retry, GHCN missing-observation guards, ISO-NE loader formats, seeded splits, and the generation/calibration logic. Pre-commit (black/flake8/mypy) green.

A pre-merge review pass (7 angles, per-finding verification) was run on the full branch diff; all findings were fixed in the final commit.

### CODE REVIEW

**Comment by doctorbrownphd** [comment]
Closing — opened prematurely; will resubmit after internal review.
